src/workflows-app/SubmissionHistory.test.js

```diff
@@ -0,0 +1,561 @@
+    expect(rows.length).toBe(3);
+
+    const headers = within(rows[0]).getAllByRole('columnheader');
+    expect(headers.length).toBe(6);
+
+    const topRowCells = (column) => {
+      const topRowCells = within(rows[1]).getAllByRole('cell');
+      return topRowCells[column];
+    };
+    // Click on "Date Submitted" column and check that the top column is correct for:
+    // * ascending order
+    await user.click(await within(headers[headerPosition['Date Submitted']]).getByRole('button'));
+    within(topRowCells(headerPosition['Date Submitted'])).getByText(/Jul 10, 2021/);
+
+    // * descending order
+    await user.click(await within(headers[headerPosition['Date Submitted']]).getByRole('button'));
+    within(topRowCells(headerPosition['Date Submitted'])).getByText(/Jan 1, 2022/);
+
+    // Click on "Status" column and check that the top column is correct for:
+    // * ascending order
+    await user.click(await within(headers[headerPosition.Status]).getByRole('button'));
+    within(topRowCells(headerPosition.Status)).getByText('Success');
+
+    // * descending order
+    await user.click(await within(headers[headerPosition.Status]).getByRole('button'));
+    within(topRowCells(headerPosition.Status)).getByText('Failed with 1 errors');
+
+    // Click on "Duration" column and check that the top column is correct for:
+    // * ascending order
+    await user.click(await within(headers[headerPosition.Duration]).getByRole('button'));
+    within(topRowCells(headerPosition.Duration)).getByText('1 day 1 hour 1 minute 1 second');
+
+    // * descending order
+    await user.click(await within(headers[headerPosition.Duration]).getByRole('button'));
+    within(topRowCells(headerPosition.Duration)).getByText('1 month 1 day 1 hour 1 minute 1 second');
+  });
+
+  it('should display no content message when there are no previous run sets', async () => {
+    // Arrange
+    const mockRunSetResponse = jest.fn(() => Promise.resolve([]));
+    const mockLeoResponse = jest.fn(() => Promise.resolve(mockAzureApps));
+
+    await Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            get: mockRunSetResponse,
+          },
+        },
+        Apps: {
+          listAppsV2: mockLeoResponse,
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      render(
+        h(BaseSubmissionHistory, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      );
+    });
+
+    // Assert
+    await waitFor(() => {
+      expect(mockRunSetResponse).toBeCalled();
+    });
+
+    const table = screen.getByRole('table');
+    expect(table).toHaveAttribute('aria-colcount', '6');
+    expect(table).toHaveAttribute('aria-rowcount', '1');
+
+    const rows = within(table).getAllByRole('cell');
+    within(rows[1]).findByText('Nothing here yet! Your previously run submissions will be displayed here.');
```

Comment:
`findBy...` queries return Promises and do need to be awaited.
https://testing-library.com/docs/queries/about#types-of-queries